### PR TITLE
Semenets/book models update

### PR DESCRIPTION
**Pull request by SemenetsSergii**

### CODE REVIEW

File: book_service/serializers.py [hunk 1]
```diff
@@ -14,3 +14,25 @@ class Meta:
...
+        fields = ("id", "title", "author", "cover", "inventory", "daily_fee", "image")
+
+
+class BookImageSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Book
+        fields = (
+            "id",
+            "image",
+        )
```
Comment (BileichukIvan) [inline book_service/serializers.py]: You should adhere to the same style of writing fields

File: book_service/tests/tests_book_views.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    def test_filter_books_by_author(self):
+        url = reverse("book_service:book-list")
+        response = self.client.get(url, {"author": "David Beazley & Brian K. Jones"})
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        # Adjust based on the actual response structure
+        data = response.data.get("results", [])  # Adjust 'results' if needed
+
```
Comment (BileichukIvan) [inline book_service/tests/tests_book_views.py]: Delete unnecessary comments

**Review by BileichukIvan — APPROVED** [review]
Good

File: book_service/views.py [hunk 3]
```diff
@@ -1,9 +1,78 @@
...
+        if self.action == "list":
+            return BookListSerializer
+
+        if self.action == "retrieve":
+            return BookDetailSerializer
+
+        if self.action == "upload_image":
+            return BookImageSerializer
+
+        return BookSerializer
```
Comment (Irina17191) [inline book_service/views.py]: Here you can return self.serializer_class, it would look more pythonic

Comment (Yuruch) [reply book_service/views.py]: Yep if you already have self.serializer then why not to use it )

File: book_service/views.py [hunk 4]
```diff
@@ -1,9 +1,78 @@
...
+        detail=True,
+        url_path="upload-image",
+        permission_classes=[IsAdminUser],
+    )
+    def upload_image(self, request, pk=None):
+        """Endpoint for uploading image to a specific book"""
+        book = self.get_object()
+        serializer = BookImageSerializer(book, data=request.data, partial=True)
+
+        if serializer.is_valid():
```
Comment (Irina17191) [inline book_service/views.py]: Here you can write serializer.is_valid(raise_exception=True)
If the serializer is not valid, an exception will be raised automatically, so there is no need to write the final return.

serializer.is_valid(raise_exception=True)
serializer.save()
return Response(serializer.data, status=status.HTTP_200_OK)

**Comment by Irina17191** [comment]
Great job!)

**Review by Irina17191 — APPROVED** [review]

**Pull request merged**